### PR TITLE
Use ``headlessFirefox`` and update to Node 20

### PR DESCRIPTION
**Pull request by AA-Turner**
I had a very old branch (https://github.com/AA-Turner/sphinx/commit/34090858fd48c7c41e9d559f4739276be09a8720 // https://github.com/AA-Turner/sphinx/commits/use-jasmine/), most of which @jayaddison covered in the recent #12754. This PR takes a few ideas from that branch, such as avoiding `xvfb-run` as we don't need it, and also updates to the latest node LTS.

A

### CODE REVIEW

**Comment by jayaddison** [comment]
Thanks for the mention @AA-Turner - one question: when I [search for `nodejs` in the Ubuntu package directory](https://packages.ubuntu.com/search?keywords=nodejs&searchon=names&suite=all&section=all) I find that Ubuntu `noble` is listed as LTS, and currently distributes NodeJS 18.19.1+dfsg-6ubuntu5.  v18 also seems to be [Node's current LTS distribution](https://nodejs.org/download/release/latest-hydrogen/) (codename `hydrogen`, v18.20.4).

Long story short: should we prefer v18 here instead of v20?  (and/or maybe make it `env.node-version`-configurable again?)

**Comment by AA-Turner** [comment]
[The website](https://nodejs.org/en/download/prebuilt-installer) lists 20 as the latest LTS, as does [endoflife.date](https://endoflife.date/nodejs). Version 18 is security-only.

> configurable again

I didn't change `env.node-version`, I only increased the version number.

A

**Comment by jayaddison** [comment]
Ok, yep - my mistake; I'd checked the NodeJS readme, but not their website.  Filed a small bugreport at https://github.com/nodejs/node/issues/54419 to hopefully clear that up.
